app/controllers/preferences_controller.rb

```diff
@@ -24,6 +26,14 @@ def create
     end
   end
 
+  def update
+    if @preference.update(preference_params)
+      redirect_to preferences_path, notice: t('views.preferences.update_success')
+    else
+      render :edit, status: :unprocessable_entity
+    end
```

Comment:
Maybe this is clear?

```suggestion
    @preference.update!(preference_form_params)

    if @preference.save
      redirect_to preferences_path, notice: t('views.preferences.update_success')
    else
      render :edit, status: :unprocessable_entity
    end
```

Reply:
mmm, I tried that but I needed to add a rescue, to catch the exception from the update!
This code also works, but I think I prefer mine 🤭

```
def update
@preference.update!(preference_params)

redirect_to preferences_path, notice: t('views.preferences.update_success')
rescue ActiveRecord::RecordInvalid
render :edit, status: :unprocessable_entity
end
```